src/target/cortexm.c

```diff
@@ -470,6 +470,10 @@ bool cortexm_probe(adiv5_access_port_s *ap)
 		ap->dp->designer_code == JEP106_MANUFACTURER_ARM_CHINA)
 		target->designer_code = JEP106_MANUFACTURER_ARM_CHINA;
 
+	/* Some STM32F103 counterfeits have a broken manufacturer code. Map to STM */
+	if (target->designer_code == JEP106_MANUFACTURER_ERRATA_CS_ASCII)
```

Comment:
Why not patch this in adiv5.c the same way as the other `JEP106_MANUFACTURER_ERRATA_CS` is (having it generate the same the debug warning)? Ideally, the MM32 patch that's done just above this would be in adiv5.c too, as this isn't really specific to Cortex-M parts but more generically ADIv5 identity handling..

Reply:
Moved to adiv5.c as suggested. I also moved the designer code re-write and the associated warning out of the `if (pidr & PIDR_JEP106_USED)` clause to avoid duplicating the warning in both _if_ and _else_-clause. Fixed a formatting mistake in the warning on the way (previously it printed a `0x` prefix followed by the manufacturer id in decimal, now it's correctly printed in hexadecimal)

I also removed my extra DEBUG_INFO and added the raw PIDR to the "ROM: " debug output as suggested above.

I decided not to move the MM32 patch from cortexm.c because of several reasons: First, after refactoring my patch would not touch cortexm.c at all, second I wasn't sure where to exactly put it in adiv5.c and most importantly I don't have that hardware available to verify that it still works and didn't want to put anything at risk here.

Reply:
Ah, good catch on that formatting bug!